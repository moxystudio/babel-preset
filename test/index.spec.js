'use strict';

const serializerPath = require('jest-serializer-path');
const preset = require('..');

expect.addSnapshotSerializer(serializerPath);

describe('targets', () => {
    it('should return a default config that target browsers and node', () => {
        expect(preset()).toMatchSnapshot();
    });

    it('should return a config for node if options.targets is set to node', () => {
        expect(preset(null, { targets: ['node'] })).toMatchSnapshot();
    });

    it('should return a config for browsers if options.targets is set to browsers', () => {
        expect(preset(null, { targets: ['browsers'] })).toMatchSnapshot();
    });

    it('should allow overriding options.targets with an object', () => {
        expect(preset(null, { targets: { node: 'current' } })).toMatchSnapshot();
    });
});

describe('modules', () => {
    it('should set modules to false if options.modules is disabled', () => {
        expect(preset(null, { modules: false })).toMatchSnapshot();
    });

    it('should set modules to false by default if BABEL_ENV is es', () => {
        process.env.BABEL_ENV = 'es';

        try {
            expect(preset(null)).toMatchSnapshot();
        } finally {
            delete process.env.BABEL_ENV;
        }
    });
});

describe('react', () => {
    it('should enable react plugins if options.react is enabled and enable development goodies', () => {
        expect(preset(null, { react: true })).toMatchSnapshot();
    });

    it('should enable react optimizations if options.env is production', () => {
        expect(preset(null, { env: 'production', react: true })).toMatchSnapshot();
    });
});

describe('lodash', () => {
    it('should disable the lodash plugin if options.lodash is disabled', () => {
        expect(preset(null, { lodash: false })).toMatchSnapshot();
    });

    it('should pass any options to the lodash plugin', () => {
        expect(preset(null, { lodash: { id: 'lodash-compat' } })).toMatchSnapshot();
    });
});

describe('namedDefaultExport & babel-plugin-add-module-exports', () => {
    it('should enable add-module-exports by default', () => {
        expect(preset(null)).toMatchSnapshot();
    });

    it('should enable add-module-exports if options.modules is commonjs', () => {
        expect(preset(null)).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs' })).toMatchSnapshot();
    });

    it('should not enable add-module-exports if options.modules is not commonjs', () => {
        expect(preset(null, { modules: false })).toMatchSnapshot();
    });

    it('should respect options.namedDefaultExport', () => {
        expect(preset(null, { namedDefaultExport: true })).toMatchSnapshot();
        expect(preset(null, { namedDefaultExport: false })).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs', namedDefaultExport: true })).toMatchSnapshot();
        expect(preset(null, { modules: 'commonjs', namedDefaultExport: false })).toMatchSnapshot();
    });

    it('should throw when enabling options.namedDefaultExport and options.modules is not commonjs', () => {
        expect(() => preset(null, { modules: false, namedDefaultExport: true }))
        .toThrow('The `namedDefaultExport` option can only be enabled when `modules` is commonjs');
    });
});
