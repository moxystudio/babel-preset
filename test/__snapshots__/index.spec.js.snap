// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lodash should disable the lodash plugin if options.lodash is disabled 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`lodash should pass any options to the lodash plugin 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": "lodash-compat",
      },
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`modules should set modules to false by default if BABEL_ENV is es 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`modules should set modules to false if options.modules is disabled 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports by default 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports if options.modules is commonjs 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should enable add-module-exports if options.modules is commonjs 2`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should not enable add-module-exports if options.modules is not commonjs 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 2`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 3`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`namedDefaultExport & babel-plugin-add-module-exports should respect options.namedDefaultExport 4`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`react should enable react optimizations if options.env is production 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-display-name/lib/index.js",
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-remove-prop-types/lib/index.js",
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-constant-elements/lib/index.js",
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-inline-elements/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`react should enable react plugins if options.react is enabled and enable development goodies 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-display-name/lib/index.js",
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-jsx-source/lib/index.js",
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-jsx-self/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`targets should allow overriding options.targets with an object 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "node": "current",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`targets should return a config for browsers if options.targets is set to browsers 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`targets should return a config for node if options.targets is set to node 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`targets should return a default config that target browsers and node 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/babel-plugin-transform-class-properties/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {},
    ],
    "babel-plugin-add-module-exports",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/babel-preset-env/lib/index.js",
      Object {
        "debug": false,
        "loose": true,
        "modules": "commonjs",
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "node": "8.9",
        },
        "useBuiltIns": true,
      },
    ],
  ],
}
`;
